src/main.rs

```diff
@@ -165,22 +180,58 @@ async fn main() {
             }
         }
 
-        // Send POI message at a fixed frequency
-        if block_number % examination_frequency == 0 {
-            compare_block = block_number + wait_block_duration;
-
-            let block: Block<_> = provider.get_block(block_number).await.unwrap().unwrap();
-            let block_hash = format!("{:#x}", block.hash.unwrap());
-
-            // Radio specific message content query function
-            // Function takes in an identifier string and make specific queries regarding the identifier
-            // The example here combines a single function provided query endpoint, current block info
-            // Then the function gets sent to agent for making identifier independent queries
-            let poi_query = partial!( query_graph_node_poi => graph_node_endpoint.clone(), _, block_hash.to_string(),block_number.try_into().unwrap());
-            let identifiers = GOSSIP_AGENT.get().unwrap().content_identifiers();
+        // Radio specific message content query function
+        // Function takes in an identifier string and make specific queries regarding the identifier
+        // The example here combines a single function provided query endpoint, current block info based on the subgraph's indexing network
+        // Then the function gets sent to agent for making identifier independent queries
+        let identifiers = GOSSIP_AGENT.get().unwrap().content_identifiers();
+        for id in identifiers {
+            // Get the indexing network of the deployment
+            let network = match query_graph_node_deployment_network(
+                graph_node_endpoint.clone(),
+                id.clone(),
+            )
+            .await
+            {
+                Ok(network) => network,
+                Err(e) => {
+                    warn!("Could not query for the subgraph's indexing network, check Graph node's indexing statuses of the deployment {}: {}", id.clone(), e);
+                    continue;
+                }
+            };
+
+            let poi_query =
+                partial!( query_graph_node_poi => graph_node_endpoint.clone(), id.clone(), _, _);
+            // get the indexing network's block provider and consensus block
+            let indexing_network_provider: &Provider<Http> = provider_endpoints.get(&network[..]).unwrap_or_else(|| panic!("Missing a block provider for the indexing network {}, please provide one named `PROVIDER_{}`", network.clone(), network.clone().to_uppercase())).as_ref().expect("Could not make construct a valid provider");
```

Comment:
Should be `panic` here? Maybe an Indexer has a lot of allocations but only wants to use/test the Radio on specific chains?

Reply:
Also maybe should be "Could not construct a valid provider"

Reply:
I think panic isn't necessary as you said that they might not want to include every allocations. In terms of urls, it wouldn't be all in the long term but should support endpoints that's complaint to Ethereum json-rpc